ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -141,14 +138,35 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
             properties);
 
         // set instrumentation key
-        telemetryItem.instrumentationKey = this._globalconfig.instrumentationKey;
-
-        var iKeyNoDashes = this._globalconfig.instrumentationKey.replace(/-/g, "");
-        telemetryItem.name = telemetryItem.name.replace("{0}", iKeyNoDashes);
+        this._setTelemetryNameAndIKey(telemetryItem);
 
         this.core.track(telemetryItem);
     }
 
+    public startTrackEvent(name: string) {
```

Comment:
Do we need those methods? I feel like those are hard to explain and nobody actually uses them

Reply:
We don't have history but I thought the intent was to allow people to do timing using a custom Event (dfferent than page view perf calculation). I agree it does not provide much value as part C data. Do you have any history in terms of customer asks/features this was added to support initially?